pkg/aws/manager_rds_subnet_group_test.go

```diff
@@ -0,0 +1,271 @@
+			},
+			wantErr: "failed to list tags for database subnet group: ",
+		},
+		{
+			name: "error when deleting subnet group fails",
+			fields: fields{
+				rdsClient: func() *rdsClientMock {
+					fakeClient, err := fakeRDSClient(func(c *rdsClientMock) error {
+						c.ListTagsForResourceFunc = func(in1 *rds.ListTagsForResourceInput) (output *rds.ListTagsForResourceOutput, e error) {
+							return &rds.ListTagsForResourceOutput{
+								TagList: []*rds.Tag{
+									{
+										Key:   aws.String(fakeRDSClientTagKey),
+										Value: aws.String(fakeClusterId),
+									},
+								},
+							}, nil
+						}
+						c.DeleteDBSubnetGroupFunc = func(in *rds.DeleteDBSubnetGroupInput) (*rds.DeleteDBSubnetGroupOutput, error) {
+							return nil, errors.New("")
+						}
+						return nil
+					})
+					if err != nil {
+						t.Fatal(err)
+					}
+					return fakeClient
+				},
+				logger: fakeLogger,
+			},
+			args: args{
+				clusterId: fakeClusterId,
+				tags:      map[string]string{},
+			},
+			wantErr: "failed to delete rds db subnet group: ",
+		},
+		{
+			name: "report empty when no subnet groups match cluster id tag",
+			fields: fields{
+				rdsClient: func() *rdsClientMock {
+					fakeClient, err := fakeRDSClient(func(c *rdsClientMock) error {
+						c.DescribeDBSubnetGroupsFunc = func(in1 *rds.DescribeDBSubnetGroupsInput) (*rds.DescribeDBSubnetGroupsOutput, error) {
+							return &rds.DescribeDBSubnetGroupsOutput{
+								DBSubnetGroups: []*rds.DBSubnetGroup{
+									{
+										DBSubnetGroupArn: aws.String(fakeARN),
+									},
+								},
+							}, nil
+						}
+						c.ListTagsForResourceFunc = func(in1 *rds.ListTagsForResourceInput) (*rds.ListTagsForResourceOutput, error) {
+							return &rds.ListTagsForResourceOutput{
+								TagList: []*rds.Tag{
+									{
+										Key:   aws.String(tagKeyClusterId),
+										Value: aws.String("not a real clusterId"),
+									},
+								},
+							}, nil
+						}
+						return nil
+					})
+					if err != nil {
+						t.Fatal(err)
+					}
+					return fakeClient
+				},
+				logger: fakeLogger,
+			},
+			args: args{
+				clusterId: fakeClusterId,
+				// we only care about the cluster id tag for this test, leave additional tags empty
+				tags:   map[string]string{},
+				dryRun: false,
+			},
+			want: make([]*clusterservice.ReportItem, 0),
+		},
+		// {
+		// 	name: "report empty when cluster id tag matches but additional tags do not",
+		// 	fields: fields{
+		// 		rdsClient: func() *rdsClientMock {
+		// 			fakeClient, err := fakeRDSClient(func(c *rdsClientMock) error {
+		// 				return nil
+		// 			})
+		// 			if err != nil {
+		// 				t.Fatal(err)
+		// 			}
+		// 			return fakeClient
+		// 		},
+		// 		logger: fakeLogger,
+		// 	},
+		// 	args: args{
+		// 		clusterId: fakeRDSClientTagVal,
+		// 		tags: map[string]string{
+		// 			"addTagKey": "addTagVal",
+		// 		},
+		// 		dryRun: true,
+		// 	},
+		// 	want: make([]*clusterservice.ReportItem, 0),
+		// },
```

Comment:
Do we want to remove this?

Reply:
It depends, it's a test around extra tags. There's some logic in the RDS manager that also matches RDS instances based on some additional tags other than the clusterID. I don't really know what the purpose of it is but I was wondering if we need to add the same thing for the RDS Subnet groups also?